### PR TITLE
Docs reorganization

### PR DESCRIPTION
**Pull request by Riksu9000**
Organized the readme.
Removed a broken article link.
Removed a few unnecessary notes.
Some flashing instructions were in separate documents and some were in buildAndProgram.md. They've all been moved to separate documents as a prerequisite for me to approve #801.
Other minor things.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,28 +26,27 @@ Fast open-source firmware for the [PineTime smartwatch](https://www.pine64.org/p
  - [Rough structure of the code](doc/code/Intro.md)
  - [How to implement an application](doc/code/Apps.md)
  - [Generate the fonts and symbols](src/displayapp/fonts/README.md)
- - [Creating a stopwatch in Pinetime(article)](https://pankajraghav.com/2021/04/03/PINETIME-STOPCLOCK.html)
  - [Tips on designing an app UI](doc/ui_guidelines.md)
-
-### InfiniSim Simulator
-Use the [InfiniSim Simulator](https://github.com/InfiniTimeOrg/InfiniSim) to experience the `InfiniTime` user interface directly on your PC, to shorten the time until you get your hands on a real [PineTime smartwatch](https://www.pine64.org/pinetime/).
-Or use it to develop new Watchfaces, new Screens, or quickly iterate on the user interface.
+ - [Bootloader, OTA and DFU](./bootloader/README.md)
```
Comment (FintasticMan) [inline README.md:30]: For consistency, there shouldn't be the `./` in front of the path.

File: README.md [hunk 2]
```diff
@@ -24,28 +26,27 @@ Fast open-source firmware for the [PineTime smartwatch](https://www.pine64.org/p
...
- - [Flash the firmware using OpenOCD and STLinkV2](doc/openOCD.md)
- - [Flash the firmware using SWD interface](doc/SWD.md)
  - [Build the project with Docker](doc/buildWithDocker.md)
  - [Build the project with VSCode](doc/buildWithVScode.md)
- - [Bootloader, OTA and DFU](./bootloader/README.md)
+ - [Flash the firmware using OpenOCD and STLinkV2](doc/openOCD.md)
+ - [Flash the firmware using SWD interface](doc/SWD.md)
+ - [Flash the firmware using JLink](doc/jlink.md)
+ - [Flash the firmware using GDB](doc/gdb.md)
  - [Stub using NRF52-DK](./doc/PinetimeStubWithNrf52DK.md)
```
Comment (FintasticMan) [inline README.md:50]: Same as above.

**Review by FintasticMan — APPROVED** [review]
Overall a good improvement.

**Review by JF002 — APPROVED** [review]
That looks good to me :+1 !
However, let's not forget that we are also working on completely revamping the documentation in #838 !

**Pull request merged**